rules/onchain_budget.go

```diff
@@ -0,0 +1,785 @@
+package rules
```

Comment:
perhaps this is a good time to start a doc where we explain small details like the memo field so that it is easy to recap in future?

so perhaps let's add a `rules/docs/onchain_budget.md` where we explain some things here?

Reply:
cool, I added a small doc, let me know if I missed anything there